mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java

```diff
@@ -160,14 +192,63 @@ public void update(CRUDOperationContext ctx,
                 }
                 docIndex++;
             }
+            // if we have any remaining items to update
+            if (numUpdating > 0) {
+                try {
+                    BulkWriteResult result = executeAndLogBulkErrors(bwo);
+                    numUpdated += result.getModifiedCount();
+                    numMatched += result.getMatchedCount();
+                } catch (Exception e) {
+                    LOGGER.warn("Update exception for documents for query: {}", query.toString());
+                }
+            }
         } finally {
             if (cursor != null) {
                 cursor.close();
             }
         }
+        handleBulkWriteError(docUpdateErrors, docUpdateAttempts);
+
+        docUpdateActual.stream().filter(doc -> doc != null).forEach(doc -> {
+            ctx.getFactory().getInterceptors().callInterceptors(InterceptPoint.POST_CRUD_UPDATE_DOC, ctx, doc);
+        });
+
         response.setNumUpdated(numUpdated);
         response.setNumFailed(numFailed);
-        response.setNumMatched(docIndex);
+        response.setNumMatched(numMatched);
+    }
+
+    private void handleBulkWriteError(List<BulkWriteError> errors, List<DocCtx> docs) {
+        docUpdateActual.addAll(docs);
+        for (BulkWriteError e : errors) {
+            DocCtx doc = docs.get(e.getIndex());
+            // remove any docs from the updated list if they had an error
+            docUpdateActual.set(e.getIndex(), null);
```

Comment:
Had initially needed to remove items from the list without affecting indexes, but removed that code before committing.  Will just remove this.